gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/domain_service/DocumentationValidationDomainService.java

```diff
@@ -52,4 +59,75 @@ public void validateContentIsSafe(String content) {
     public void validateTemplate(String pageContent, String apiId) {
         this.templateResolverDomainService.resolveTemplate(pageContent, Map.of("api", this.apiCrudService.get(apiId)));
     }
+
+    public void checkMarkdownOrSwaggerConsistency(Page page) {
+        PageSituation newPageParentSituation = getPageSituation(page.getParentId());
+        if (newPageParentSituation == PageSituation.SYSTEM_FOLDER || newPageParentSituation == PageSituation.IN_SYSTEM_FOLDER) {
```

Comment:
For v4 API documentation, there is no notion of "System folder" and we are not handling Translations. What do you think of removing this + `getPageSituation` and restricting page creation to only V4 APIs?